misc/sync-typeshed.py

```diff
@@ -53,12 +67,17 @@ def main() -> None:
     if answer.lower() != 'y':
         sys.exit('Aborting')
 
-    # TODO: Clone typeshed repo if no directory given
-    assert args.typeshed_dir
-    update_typeshed(args.typeshed_dir)
+    if not args.typeshed_dir:
+        # Clone typeshed repo if no directory given.
+        with tempfile.TemporaryDirectory() as tempdir:
+            print('Cloning typeshed in {}...'.format(tempdir))
+            subprocess.run(['git', 'clone', 'https://github.com/python/typeshed.git'],
+                           check=True, cwd=tempdir)
+            repo = os.path.join(tempdir, 'typeshed')
+            commit = update_typeshed(repo, args.commit)
+    else:
+        commit = update_typeshed(args.typeshed_dir, args.commit)
```

Comment:
I think you can simplify this a bit as
```python
if args.typeshed_dir:
    repo = args. typeshed_dir
else:
    ...
    repo = os.path.join(tempdir, 'typeshed')
commit = update_typeshed(repo, args.commit)
```
But this is really minor.

Reply:
Because the context manager deletes the temp directory on exit, the `update_typeshed` call must be within the context manager.

Reply:
Oooh, I see.